flocx_market/api/app.py

```diff
@@ -1,19 +1,46 @@
 import pecan
-from pecan import make_app
-from flocx_market import model
-from flocx_market.api import config as api_config
+from oslo_context import context
+from pecan import hooks
+import flocx_market.conf
+
+CONF = flocx_market.conf.CONF
+
+class ContextHook(hooks.PecanHook):
+    def before(self, state):
+        ctx = context.RequestContext.from_environ(state.request.environ)
+        state.request.context = ctx
+
+    def after(self, state):
+        state.request.context = None
 
 def get_pecan_config():
-    filename = api_config.__file__.replace('.pyc', '.py')   # get the absolute path of the pecan config.py
-    return pecan.configuration.conf_from_file(filename)
+    cfg_dict = {
+        "app": {
+            "root": CONF.pecan.root,
+            "modules": CONF.pecan.modules,
+            "debug": CONF.pecan.debug,
+            "auth_enable": CONF.pecan.auth_enable
+        }
+    }
+
+    return pecan.configuration.conf_from_dict(cfg_dict)
+
+def setup_app(config=None):
+    if not config:
+        config = get_pecan_config()
+
+    pecan.configuration.set_config(dict(config), overwrite=True)
 
-def setup_app():
-    config = get_pecan_config()
-    model.init_model()
-    app_conf = dict(config.app)
     app = pecan.make_app(
-        app_conf.pop('root'),
-        logging=getattr(config, 'logging', {}),
-        **app_conf)
+        config.app.root,
+        hooks=lambda: [ContextHook()],
+        debug=CONF.pecan.debug,
+        static_root=config.app.static_root if CONF.pecan.debug else None,
+        force_canonical=getattr(config.app, 'force_canonical', True),
+    )
+    # keystone authentication:
+    # if CONF.pecan.auth_enable:
+    #     app = auth_token.AuthProtocol(app, dict(CONF.keystone_authtoken))
+
     return app
 
```

Comment:
These changes probably go in a separate PR, unless they're necessary with the service changes?

Reply:
Ah, I see - it's because of the conf changes!

Reply:
For service changes, I modified the configs into a conf folder. So I change the api.py accordingly. Probably some of the changes are not necessary, I'll fix it.